File: app/lib/src/models/zone.dart
class Zone {
  const Zone({
    required this.id,
    required this.name,
    required this.slug,
    required this.city,
    required this.distanceKm,
    required this.sedanFare,
    required this.suvFare,
  });

  factory Zone.fromJson(Map<String, dynamic> json) => Zone(
        id: json['id'] as int,
        name: json['name'] as String,
        slug: json['slug'] as String? ?? '',
        city: json['city'] as String? ?? 'Bengaluru',
        distanceKm: json['distance_km'] as int? ?? 0,
        sedanFare: json['sedan_fare'] as int? ?? 0,
        suvFare: json['suv_fare'] as int? ?? 0,
      );

  final int id;
  final String name;
  final String slug;
  final String city;
  final int distanceKm;
  final int sedanFare;
  final int suvFare;
}

File: api/app/Models/Zone.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable(['city', 'name', 'slug', 'lat', 'lng', 'distance_km', 'sedan_fare', 'suv_fare', 'sort_order', 'is_active'])]
class Zone extends Model
{
    use HasFactory;

    protected function casts(): array
    {
        return [
            'lat' => 'float',
            'lng' => 'float',
            'is_active' => 'boolean',
        ];
    }

    public function scopeActive(Builder $query): Builder
    {
        return $query->where('is_active', true);
    }

    /**
     * Rides heading here that a stranger could still take a seat in.
     *
     * Exists as a relation rather than a query in the controller so the area
     * list can `withCount` it — the alternative is a count per zone on the
     * app's first screen.
     */
    public function openRides(): HasMany
    {
        return $this->hasMany(RideGroup::class)->joinable();
    }
}
